validator/src/admin_rpc_service.rs

```diff
@@ -303,6 +349,39 @@ impl AdminRpc for AdminRpcImpl {
         AdminRpcImpl::set_identity_keypair(meta, identity_keypair, require_tower)
     }
 
+    fn set_relayer_config(
+        &self,
+        meta: Self::Metadata,
+        relayer_url: String,
+        trust_packets: bool,
+        expected_heartbeat_interval_ms: u64,
+        max_failed_heartbeats: u64,
+    ) -> Result<()> {
+        debug!("set_relayer_config request received");
+
+        if relayer_url.contains("http") || relayer_url.is_empty() {
+            meta.with_post_init(|post_init| {
+                let expected_heartbeat_interval =
+                    Duration::from_millis(expected_heartbeat_interval_ms);
+
+                let oldest_allowed_heartbeat =
+                    Duration::from_millis(max_failed_heartbeats * expected_heartbeat_interval_ms);
+
+                *post_init.relayer_config.lock().unwrap() = RelayerConfig {
+                    relayer_url,
+                    expected_heartbeat_interval,
+                    oldest_allowed_heartbeat,
```

Comment:
ugh why do we have this var? how is it diff from expected_heartbeat_interval

Reply:
not you, just messy from first impl

Reply:
i can cleanup here, only minor scope creep

Reply:
its ok, can leave it for now. just me being grumpy old man lol